skycatalogs/catalog_creator.py

```diff
@@ -990,6 +1025,9 @@ def _create_pointsource_flux_pixel(self, pixel):
         # For schema use self._ps_flux_schema
         # output_template should be derived from value for flux_file_template
         #  in main catalog config.  Cheat for now
+
+        global _star_collection
```

Comment:
I'm not positive it does need to be global.  The star code is following the same procedure as the galaxy code multiprocessing code, which was written a while ago.  My intent then was to make the rather large _star_collection available to subprocesses without pickling.  I don't know whether that is in fact the case.
If it's not I would do better to just pass in the slice of _star_collection that each subprocess needs, but that is not something I want to take on for this PR.

Reply:
It seems like you would need to omit `_star_collection` from the argument list and declare it global inside of `_do_star_flux_chunk` to avoid having it pickled.  Given that it's being passed as an argument, I think it must be being serialized, and the global declaration here isn't doing anything.  If so, then there's no reason to pass `l_bnd` and `u_bnd` and the slicing can be done in the calling code. 

> that is not something I want to take on for this PR.

The code may work fine as-is, these apparent inconsistencies seem like good reasons to try to understand what's going on in case there is something untoward actually happening.  So if you don't want to change it, I'd suggest at least some test code to ensure it is behaving as expected.

Reply:
I agree in principle; packaging this up in test code will take some thought. When I first implemented the identical scheme for galaxies I proceeded pretty carefully by
* examining whatever I could in the debugger 
* comparing output from creating a flux file with number-of-subprocesses = 1 (in which case the code doesn't use subprocesses at all and just calls the `_do_X_flux_chunk` routine directly) and number-of-subprocesses set to something realistic, like 20.   The outputs were identical.

I did the same thing when I implemented parallel processing for stars. I haven't done the second test for this last commit, but the code changes have nothing to do with the parallel processing part of the code.

Reply:
Sure, but statements like this

> My intent then was to make the rather large _star_collection available to subprocesses without pickling. I don't know whether that is in fact the case.

make it rather unclear to me, without actually having run the code (which apparently isn't that easy to do), that the unmodified implementation is correct.  Regardless of whether consistent outputs are obtained with 1 or more than 1 subprocess, it would be nice to have the global vs not global aspects of the code make sense.   Right now, they don't appear to me to be consistent.  Given that it looks like `global _star_collection` isn't doing what's intended, it would be remiss of me not to point that out and suggest a fix.